### PR TITLE
[kube-state-metrics] Add support for global imagePullSecrets values

### PR DESCRIPTION
**Pull request by maartengo**
#### What this PR does / why we need it
Some parent charts of kube-state-metrics have global values. As a user it is confusing when global values aren't consistently used across charts.

#### Which issue this PR fixes
This PR adds support for global `imagePullSecrets`. This makes it easier for users of charts such as kube-prometheus-stack to configure custom pull secrets.

#### Special notes for your reviewer
This is my first PR to the prometheus-community. Any feedback is greatly appreciated.

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/kube-state-metrics/templates/_helpers.tpl [hunk 1]
```diff
@@ -99,3 +99,28 @@ labelNameLengthLimit: {{ . }}
 labelValueLengthLimit: {{ . }}
 {{- end }}
 {{- end -}}
+
+{{/*
+To help compatibility with other charts which use global.imagePullSecrets.
```
Comment (monotek) [inline charts/kube-state-metrics/templates/_helpers.tpl:104]: please move description to values.yaml.

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/kube-state-metrics/templates/_helpers.tpl (hunk 1 above)
Comment (maartengo) [reply charts/kube-state-metrics/templates/_helpers.tpl]: Moved! Also added clarification about the required input for the template.

**Review by dotdc — APPROVED** [review]
LGTM

**Review by zanhsieh — APPROVED** [review]

**Comment by zanhsieh** [comment]
@maartengo 
Thank you for the contribution. Would you mind to bump the version in Chart.yaml to latest please?

**Review by zanhsieh — CHANGES_REQUESTED** [review]
@maartengo
Thank you for the contribution. Would you mind to bump the version in Chart.yaml to latest please?

File: charts/kube-state-metrics/templates/serviceaccount.yaml [hunk 2]
```diff
@@ -11,5 +11,5 @@ metadata:
 {{ toYaml .Values.serviceAccount.annotations | indent 4 }}
 {{- end }}
 imagePullSecrets:
-{{ toYaml .Values.serviceAccount.imagePullSecrets | indent 2 }}
+{{- include "kube-state-metrics.imagePullSecrets" (dict "Values" .Values "imagePullSecrets" .Values.serviceAccount.imagePullSecrets) | indent 2 }}
```
Comment (zanhsieh) [inline charts/kube-state-metrics/templates/serviceaccount.yaml:14]:
```suggestion
  {{- include "kube-state-metrics.imagePullSecrets" (dict "Values" .Values "imagePullSecrets" .Values.serviceAccount.imagePullSecrets) | indent 2 }}
```

File: charts/kube-state-metrics/templates/deployment.yaml [hunk 3]
```diff
@@ -146,9 +146,9 @@ spec:
         securityContext:
 {{ toYaml .Values.containerSecurityContext | indent 10 }}
 {{- end }}
-{{- if .Values.imagePullSecrets }}
+{{- if or .Values.imagePullSecrets .Values.global.imagePullSecrets }}
       imagePullSecrets:
-{{ toYaml .Values.imagePullSecrets | indent 8 }}
+{{- include "kube-state-metrics.imagePullSecrets" (dict "Values" .Values "imagePullSecrets" .Values.imagePullSecrets) | indent 8 }}
```
Comment (zanhsieh) [inline charts/kube-state-metrics/templates/deployment.yaml:151]:
```suggestion
        {{- include "kube-state-metrics.imagePullSecrets" (dict "Values" .Values "imagePullSecrets" .Values.imagePullSecrets) | indent 8 }}
```

**Review by zanhsieh — CHANGES_REQUESTED** [review]

**Review by zanhsieh — APPROVED** [review]

**Comment by maartengo** [comment]
Is there anything I can do to get this merged? The PR seems blocked on a change request from @monotek, but so far as I can see its already fulfilled.

**Comment by zanhsieh** [comment]
@maartengo 
Either @monotek or @dotdc has to re-approve again.

**Review by dotdc — APPROVED** [review]
LGTM

**Comment by dotdc** [comment]
Done, thank you for your contribution @maartengo !

**Pull request merged**